src/dxtbx/model/scan.h

```diff
@@ -155,144 +197,231 @@ namespace dxtbx { namespace model {
     scitbx::af::shared<double> get_epochs() const {
-      return epochs_;
+      DXTBX_ASSERT(properties_.contains("epochs"));
+      return properties_.get<double>("epochs");
     }
 
-    /** Set the image range */
     void set_image_range(vec2<int> image_range) {
+      /*
+       * Known issue with this function
+       * https://github.com/cctbx/dxtbx/issues/497
+       */
       image_range_ = image_range;
       num_images_ = 1 + image_range_[1] - image_range_[0];
-      epochs_.resize(num_images_);
-      exposure_times_.resize(num_images_);
+      properties_.resize(num_images_);
+
+      // Edge case where num_images_ was 1 and has been increased
+      if (properties_.contains("oscillation_width") && num_images_ > 1) {
+        scitbx::af::shared<double> osc_width =
+          properties_.get<double>("oscillation_width");
+
+        vec2<double> osc = vec2<double>(get_oscillation()[0], osc_width[0]);
+        set_oscillation(osc);
+
+        // oscillation_width only needed when num_images_ == 1
+        dxtbx::af::flex_table_suite::delitem_column(properties_, "oscillation_width");
+      }
       DXTBX_ASSERT(num_images_ > 0);
     }
 
-    /** Set the batch_offset */
     void set_batch_offset(int batch_offset) {
       batch_offset_ = batch_offset;
     }
 
-    /** Set the oscillation */
     void set_oscillation(vec2<double> oscillation) {
-      DXTBX_ASSERT(oscillation[1] >= 0.0);
-      oscillation_ = oscillation;
+      scitbx::af::shared<double> oscillation_arr(num_images_);
+      for (std::size_t i = 0; i < num_images_; ++i) {
+        oscillation_arr[i] = oscillation[0] + oscillation[1] * i;
+      }
+      set_property("oscillation", oscillation_arr.const_ref());
+
+      // Edge case where the oscillation width needs to be stored
+      if (num_images_ == 1) {
+        scitbx::af::shared<double> oscillation_width_arr(num_images_);
+        oscillation_width_arr[0] = oscillation[1];
+        set_property("oscillation_width", oscillation_width_arr.const_ref());
+      }
     }
 
-    /** Set the exposure time */
     void set_exposure_times(scitbx::af::shared<double> exposure_times) {
       DXTBX_ASSERT(exposure_times.size() == num_images_);
-      exposure_times_ = exposure_times;
+      set_property("exposure_time", exposure_times.const_ref());
+      DXTBX_ASSERT(properties_.is_consistent());
     }
 
-    /** Set the image epochs */
     void set_epochs(const scitbx::af::shared<double> &epochs) {
       DXTBX_ASSERT(epochs.size() == num_images_);
-      epochs_ = epochs;
+      set_property("epochs", epochs.const_ref());
+      DXTBX_ASSERT(properties_.is_consistent());
     }
 
     /** Get the total oscillation range of the scan */
     vec2<double> get_oscillation_range() const {
-      return vec2<double>(oscillation_[0],
-                          oscillation_[0] + num_images_ * oscillation_[1]);
+      vec2<double> osc = get_oscillation();
+      return vec2<double>(osc[0], osc[0] + num_images_ * osc[1]);
```

Comment:
e.g. here, isn't the actual answer _literally_ `osc[0], osc[num_images_-1]`, even if the user has set per-image oscillations?